### ISSUE TITLE
vips copy fails with `pngload: non-recoverable state`

### ISSUE THREAD

**Issue by adriaanmeuris** [libvips/libvips#2170]
**Describe the bug**
any operation with the attached png fails with error `pngload: non-recoverable state`.
[input.png.zip](https://github.com/libvips/libvips/files/6220696/input.png.zip)

**To Reproduce**
Use the attached image and run:
```
vips copy input.png output.png
```

**Expected behavior**
file is copied

**Actual behavior**
Error is thrown:
```
pngload: non-recoverable state
vips2png: unable to write to target output.png
```

**Environment**
(please complete the following information)
 - OS: MacOS 11.2.3
 - Vips: 8.10.5 installed through Homebrew

**Comment by jcupitt** [comment]
Hi @adriaanmeuris,

This should be fixed in 8.10.6. I see:

```
john@banana ~/pics $ vips --version
vips-8.10.6-Mon Mar 29 11:05:11 UTC 2021
john@banana ~/pics $ vips copy input.png x.ppm

(vips:3945227): VIPS-WARNING **: 12:11:15.926: gAMA: gamma value out of range

(vips:3945227): VIPS-WARNING **: 12:11:15.926: gAMA: gamma value out of range

(vips:3945227): VIPS-WARNING **: 12:11:15.929: gAMA: gamma value out of range
```

Hopefully homebrew will update soon.

**Comment by adriaanmeuris** [comment]
great to hear, thanks for the speedy reply!
